ecommerce/serializers.py

```diff
@@ -56,3 +58,37 @@ class Meta:
             'price',
             'description',
         )
+
+
+class CheckoutDataSerializer(serializers.ModelSerializer):
+    """Serializer for ecommerce information for a BootcampApplication"""
+    bootcamp_run = BootcampRunSerializer()
+    total_price = serializers.DecimalField(max_digits=20, decimal_places=2, read_only=True, source="price")
+    total_paid = serializers.DecimalField(max_digits=20, decimal_places=2, read_only=True)
+    payments = serializers.SerializerMethodField()
+    installments = serializers.SerializerMethodField()
+
+    def get_payments(self, application):
+        """Serialized payments made by the user"""
+        return LineSerializer(
+            (order.line_set.first() for order in application.orders.filter(status=Order.FULFILLED)),
```

Comment:
If we change this to `(order.line_set.first() for order in application.orders.all() if order.status == Order.FULFILLED)`, that will take advantage of the prefetch. I think this will generate another query with that `filter()`.

Reply:
Also I think we can prefetch `orders__line_set` in `get_queryset` instead of just `orders`

Reply:
Sounds good